WebLogic JMS Transaction Timeout Triggers Exponential Backoff Instead of Session Recovery in WSO2 MI

### Description

Hi team,

The following error can be seen repeatedly logged in WSO2 MI in when integrated with WebLogic server after an transaction timeout.

```
[2026-03-03 17:01:02,277] ERROR {ServiceTaskManager} - Error receiving message for service : WeblogicJMSConsumerProxy weblogic.jms.common.TransactionRolledBackException: Attempt to resume an inactive transaction: BEA1-0017EA1D9B617F28EDB9:error resuming transacted session's internal transaction
	at weblogic.jms.dispatcher.DispatcherAdapter.convertToJMSExceptionAndThrow(DispatcherAdapter.java:110)
	at weblogic.jms.dispatcher.DispatcherAdapter.dispatchSyncNoTran(DispatcherAdapter.java:61)
	at weblogic.jms.client.JMSSession.receiveMessage(JMSSession.java:894)
	at weblogic.jms.client.JMSConsumer.receiveInternal(JMSConsumer.java:647)
	at weblogic.jms.client.JMSConsumer.receive(JMSConsumer.java:526)
	at weblogic.jms.client.WLConsumerImpl.receive(WLConsumerImpl.java:184)
	at org.apache.axis2.transport.jms.ServiceTaskManager$MessageListenerTask.receiveMessage(ServiceTaskManager.java:714)
	at org.apache.axis2.transport.jms.ServiceTaskManager$MessageListenerTask.run(ServiceTaskManager.java:519)
	at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: weblogic.jms.common.TransactionRolledBackException: Attempt to resume an inactive transaction: BEA1-0017EA1D9B617F28EDB9:error resuming transacted session's internal transaction
	at weblogic.jms.frontend.FESession.transactedException(FESession.java:2149)
	at weblogic.jms.frontend.FESession.throwTransactedException(FESession.java:2164)
	at weblogic.jms.frontend.FESession.transactedInfect(FESession.java:2272)
	at weblogic.jms.frontend.FESession.transactedInfect(FESession.java:2174)
	at weblogic.jms.frontend.FEConsumer.receive(FEConsumer.java:597)
	at weblogic.jms.frontend.FEConsumer.invoke(FEConsumer.java:864)
	at weblogic.messaging.dispatcher.Request.wrappedFiniteStateMachine(Request.java:961)
	at weblogic.messaging.dispatcher.DispatcherServerRef.invoke(DispatcherServerRef.java:276)
	at weblogic.messaging.dispatcher.DispatcherServerRef.handleRequest(DispatcherServerRef.java:141)
	at weblogic.messaging.dispatcher.DispatcherServerRef.access$000(DispatcherServerRef.java:34)
	at weblogic.messaging.dispatcher.DispatcherServerRef$2.run(DispatcherServerRef.java:111)
	at weblogic.work.ExecuteThread.execute(ExecuteThread.java:256)
	at weblogic.work.ExecuteThread.run(ExecuteThread.java:221)
Caused by: javax.transaction.InvalidTransactionException: Attempt to resume an inactive transaction: 
```

### Steps to Reproduce


1. Create a queue in WebLogic .
2. Configure a WebLogic JMS consumer proxy in WSO2 MI[1]. 
3. Set the connection factory transaction timeout to 5 seconds
4. (Domain → Services → Messaging → JMS Modules → Connection Factory Name → Transactions).
5. Start both applications.
6. After approximately 5 seconds, the TransactionRolledBackException begins to occur repeatedly.

[1] [https://mi.docs.wso2.com/en/4.2.0/install-and-setup/setup/brokers/configure-with-weblogic/](https://mi.docs.wso2.com/en/4.2.0/install-and-setup/setup/brokers/configure-with-weblogic/)

### Version

4.0.0

### Environment Details (with versions)

_No response_

--- a/modules/jms/src/main/java/org/apache/axis2/transport/jms/ServiceTaskManager.java
+++ b/modules/jms/src/main/java/org/apache/axis2/transport/jms/ServiceTaskManager.java
@@ -41,6 +41,7 @@ import javax.jms.JMSException;
 import javax.jms.Message;
 import javax.jms.MessageConsumer;
 import javax.jms.Session;
+import javax.jms.TransactionRolledBackException;
 import javax.naming.Context;
 import javax.naming.InitialContext;
 import javax.naming.NamingException;
@@ -765,6 +766,9 @@ public class ServiceTaskManager {
                     consumerRetryCount = 0;
                     return msg;
                 }
+            } catch (TransactionRolledBackException e) {
+                log.warn("Session transaction has been rolled back. Recreating JMS session.");
+                recreateSession();
             } catch (IllegalStateException e) {
                 // probably the consumer (shared) was closed.. which is still ok.. as we didn't read
                 connectionReceivedError = true;
@@ -788,6 +792,26 @@ public class ServiceTaskManager {
                 logError("Error receiving message for service : " + serviceName, e);
             }
             return null;
+        }
+
+        private synchronized void recreateSession() {
+            if (log.isDebugEnabled()) {
+                log.debug("Closing dead JMS consumer and session.");
+            }
+
+            if (consumer != null) {
+                closeConsumer(true);
+            }
+            if (session != null) {
+                closeSession(true);
+            }
+
+            session  = createSession();
+            consumer = createConsumer();
+
+            if (log.isDebugEnabled()) {
+                log.debug("JMS session recreated successfully.");
+            }
         }
 
         /**
@@ -1355,7 +1379,7 @@ public class ServiceTaskManager {
             }
 
             try {
-                context = getInitialContext();
+                context = getInitialContext(jmsProperties);
                 return
                     JMSUtils.lookup(context, UserTransaction.class, getUserTransactionJNDIName());
             } catch (NamingException e) {
@@ -1366,7 +1390,7 @@ public class ServiceTaskManager {
         
         if (sharedUserTransaction == null) {
             try {
-                context = getInitialContext();
+                context = getInitialContext(jmsProperties);
                 sharedUserTransaction =
                     JMSUtils.lookup(context, UserTransaction.class, getUserTransactionJNDIName());
                 if (log.isDebugEnabled()) {

